src/typ.rs

```diff
@@ -314,16 +314,20 @@ pub enum PatternConstructor {
 }
 
 #[derive(Debug, Clone)]
-pub struct Env<'a, 'b> {
+pub struct Typer<'a, 'b> {
     current_module: &'b [String],
     uid: usize,
+    level: usize,
     annotated_generic_types: im::HashSet<usize>,
     importable_modules: &'a HashMap<String, Module>,
     imported_modules: HashMap<String, Module>,
 
     // Values defined in the current function (or the prelude)
     local_values: im::HashMap<String, ValueConstructor>,
 
+    // Type variables in the current scope
+    type_vars: im::HashMap<String, (usize, Arc<Type>)>,
```

Comment:
There will be other type vars in the scope that are not in this hashmap- this is just the type vars from annotations. Perhaps `annotation_type_vars` would be a better name?